compiler/sem/sempass2.nim

```diff
@@ -273,12 +273,13 @@ proc warnAboutGcUnsafe(n: PNode; conf: ConfigRef) =
 proc markGcUnsafe(a: PEffects; reason: PSym) =
   if not a.inEnforcedGcSafe:
     a.gcUnsafe = true
-    if a.owner.kind in routineKinds: a.owner.gcUnsafetyReason = reason
+    if a.owner.kind in routineKinds - {skMacro}:
```

Comment:
Should happen separately from this PR, but it might be time to start thinking of a new constant name, like `routineKindsWithGC` to absorbe some of the `- {skMacro}`.

Reply:
Yeah, I agree. When doing so, it might make sense to also exclude `skTemplate` from the set -- templates don't have a GC unsafety reason nor are they transformed.